dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/FlowReduxStateMachine.kt

```diff
@@ -43,14 +42,10 @@ public abstract class FlowReduxStateMachine<S : Any, A : Any>(
             .receiveAsFlow()
             .reduxStore(logger, initialState, specBlock)
             .onStart {
-                activeFlowCounterMutex.withLock {
-                    activeFlowCounter++
-                }
+                activeFlowCounter.value++
```

Comment:
Are you sure that this doesn't just execute the same as
```
val temp = activeFlowCounter.value
activeFlowCounter.value = temp + 1
```
?

(The whole point of the mutex there is to avoid the race condition.)

And if you're not sure, may I suggest using
```
activeFlowCounter.incrementAndGet()
```
instead (yes we don't care about the result - it's about the atomic increment).

Reply:
I thought about this myself and just didn't take incrementAndGet because of the 'andGet' part. But I guess you're perfectly right in being more careful. Thanks